.github/CONTRIBUTING.md

```diff
@@ -86,7 +121,43 @@ The communities behind these PHP frameworks/CMSes/polyfill libraries are strongl
+        For example, if a sniff is looking for a new modifier keyword being allowed for OO properties, property declarations with modifiers _other_ than the one the sniff is looking for would fall into this category.
+    3. "Undetermined/ignored by the sniff" - code for which PHPCS is not typically suitable to determine whether the code is cross-version compatible or not.
+        For example, if a sniff is looking for a particular parameter type being passed to a function call, the sniff can flag this if the parameter is passed as a hard-coded value, like `10` or `false`, but the sniff cannot determine what the return type is of a function call, or find the type of a class constant declared elsewhere. In that case, the sniff should stay silent.
+    4. "The issue" - code which should be flagged by the sniff for having a PHP cross-version compatibility issue.
+        Please make sure the code snippets in this category use plenty of variation in whitespace, comment use and otherwise variation in how code can be written.
+        Sniffs should not expect to be run on "clean" codebases, so make sure to include code snippets written in the most horrible and most unreadable way you can imagine to make sure the sniff is properly tested.
+2. A test class for use with PHPUnit.
+    This test class should extend the `PHPCompatibility\Tests\BaseSniffTestCase` class which will handle triggering PHP_CodeSniffer to run the sniff over the test case file with various `testVersion` settings.
+    Typically, the test class will have three (types of) test methods:
+    1. `testTheIssueTheSniffShouldFind($line)` with a data provider passing the line numbers in the test case file on which errors/warnings should be found coming from this sniff.
+        The `testVersion` used in this test method should be one on which the sniff should throw an error/warning.
+        This method tests the snippets from snippet category 4.
+    2. `testNoFalsePositives($line)` with a data provider passing the line number in the test case file on which we explicitly do **not** expect errors/warnings to be found from this sniff.
+        This test should normally use the same `testVersion` setting as the `testTheIssueTheSniffShouldFind()` method and verifies that the sniff does not throw false positives for code in snippet categories 1, 2 and 3.
+    3. `testNoViolationsInFileOnValidVersion()` with a `$this->assertNoViolation($file);` assertion to ensure the sniff does not throw false positives with a `testVersion` on which the issue doesn't exist.
+
+Example:
+For a sniff which detects a new PHP feature, the first two test methods would set a `testVersion` of the last PHP version _before_ the new PHP feature was introduced and the third `testNoViolationsInFileOnValidVersion()` method, would set a `testVersion` of the PHP version in which the PHP feature was introduced and therefore available.
```

Comment:
```suggestion
For a sniff which detects a new feature in PHP 8.5:
- The first two test methods would set a `testVersion` of 8.4, the version _before_ the feature was introduced.
- The third `testNoViolationsInFileOnValidVersion()` method would set a `testVersion` of 8.5, the version in which the PHP feature was introduced and therefore available.
```
The sentence was getting long. I propose to split it into bullets, cut out a few redundant words, and organize the bullets in the same way (methods, version number, what this version means in context).